app/src/activity.c

```diff
@@ -69,7 +73,18 @@ void activity_work_handler(struct k_work *work) {
     int32_t current = k_uptime_get();
     int32_t inactive_time = current - activity_last_uptime;
 #if IS_ENABLED(CONFIG_ZMK_SLEEP)
-    if (inactive_time > MAX_SLEEP_MS && !is_usb_power_present()) {
+    if (inactive_time > MAX_SLEEP_MS && !is_usb_power_present()
+#if IS_ENABLED(CONFIG_ZMK_NO_SLEEP_WHILE_BLE_CONNECTED)
+    /* if user inactive and USB is not connected,
+     * keyboard will sleep as soon as BLE is disconnected
+     */
+#if !IS_ENABLED(CONFIG_ZMK_SPLIT) || IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
+        && !zmk_ble_active_profile_is_connected()
+#else
+        && !zmk_split_bt_peripheral_is_connected()
+#endif
+#endif
+    ) {
```

Comment:
Wrapping this `if` with these preprocessor checks like this feels really brittle/painful.

Thoughts on something like:

```
bool prevent_sleep = is_usb_power_present();
#if IS_ENABLED(CONFIG_ZMK_NO_SLEEP_WHILE_BLE_CONNECTED)
prevent_sleep = prevent_sleep ||
#if !IS_ENABLED(CONFIG_ZMK_SPLIT) || IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
   zmk_ble_active_profile_is_connected();
#else
   zmk_split_bt_peripheral_is_connected();
#endif
#endif
```

Then the check can be a simple `if (inactive_time > MAX_SLEEP_MX && !prevent_sleep) {`

Thoughts?